artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTUtil.java

```diff
@@ -149,16 +154,52 @@ public static void logMessage(MQTTSessionState state, MqttMessage message, boole
          if (message.fixedHeader() != null) {
             log.append(message.fixedHeader().messageType().toString());
 
-            if (message.variableHeader() instanceof MqttPublishVariableHeader) {
-               log.append("(" + ((MqttPublishVariableHeader) message.variableHeader()).messageId() + ") " + message.fixedHeader().qosLevel());
-            } else if (message.variableHeader() instanceof MqttMessageIdVariableHeader) {
+            if (message.variableHeader() instanceof MqttMessageIdVariableHeader) {
                log.append("(" + ((MqttMessageIdVariableHeader) message.variableHeader()).messageId() + ")");
             }
 
-            if (message.fixedHeader().messageType() == MqttMessageType.SUBSCRIBE) {
-               for (MqttTopicSubscription sub : ((MqttSubscribeMessage) message).payload().topicSubscriptions()) {
-                  log.append("\n\t" + sub.topicName() + " : " + sub.qualityOfService());
-               }
+            switch (message.fixedHeader().messageType()) {
+               case PUBLISH:
+                  MqttPublishVariableHeader publishHeader = (MqttPublishVariableHeader) message.variableHeader();
+                  log.append("(" + publishHeader.packetId() + ")")
+                     .append(" topic=" + publishHeader.topicName())
+                     .append(", qos=" + message.fixedHeader().qosLevel())
+                     .append(", retain=" + message.fixedHeader().isRetain())
+                     .append(", dup=" + message.fixedHeader().isDup());
+                     //.append(" payload=" + ((MqttPublishMessage)message).payload().toString(StandardCharsets.UTF_8));
```

Comment:
ditto as per other comment.
"rather than commented out should this just be removed."

Reply:
I disagree that this should be removed as it can serve as a really convenient way to look at the payload granted they're UTF-8 encoded strings.  The only reason it's commented out is because the MQTT spec has no rules on the format of the payload.  It's just an array of bytes.  However, if you happen to be working with UTF-8 encoded string then this might prove helpful.

Reply:
Yes I agree that is useful if you are using UTF8 encoded string as payload. Is it possible having this logging line as configurable (disabled by default) so that people using Strings can enable it for showing the payload.

Reply:
If that's the case would configuration to turn it on / off be better than requiring a complete rebuild in order to use them once uncommented ? 

Sidebar is there a gating mechanism on the logger being used to check if it's even enabled, all those append calls for something that might not actually be turned on seems less than ideal.

Reply:
The method starts with:

if (logger.isTraceEnabled())

So none of this will be executed if TRACE (or equivalent) isn't turned on in the logging configuration.

Reply:
And there is no way for making one of them configurable as well ? For example printing the PUBLISH payload ?

Reply:
My only thought here is to use something really basic like a system property or something.  I don't think this is worth changing the configuration schema to support a new option in broker.xml.